File: all_sort/go/insert_sort.go
package main

import "fmt"

func insert_sort(seq []int) {
	l := len(seq)
	for i := 1; i < l; i++ {
		pos := i
		value := seq[pos]
		for {
			if pos > 0 && seq[pos-1] > value {
				seq[pos] = seq[pos-1]
				pos--
			} else {
				break
			}
		}
		seq[pos] = value
	}

}

func main() {
	var lst = []int{3, 1, 2, 5, 6, 7, 9, 0}
	insert_sort(lst)
	fmt.Printf("lst: %v\n", lst)
}

File: all_sort/python/insert_sort.py
#!/usr/bin/python3
# 插入排序：使局部有序，将元素插入有序的序列里
def insert_sort(seq):
    n = len(seq)
    for i in range(1, n):
        value = seq[i]
        pos = i
        while pos > 0 and seq[pos-1] > value:
            seq[pos] = seq[pos-1]
            pos -= 1
        seq[pos] = value
import random
lst = list(range(10))
random.shuffle(lst)
print(lst)
insert_sort(lst)
print(lst)
